src/Microsoft.Azure.WebJobs.Protocols/FunctionDescriptor.cs

```diff
@@ -35,5 +36,31 @@ public class FunctionDescriptor
         /// </summary>
         [JsonIgnore]
         internal MethodInfo Method { get; set; }
+
+        /// <summary>
+        /// Gets the <see cref="TriggerParameterDescriptor"/> for this function
+        /// </summary>
+        [JsonIgnore]
+        internal TriggerParameterDescriptor TriggerDescriptor { get; set; }
+#if PUBLICPROTOCOL
+#else
```

Comment:
Two questions on this:

1. What is this public protocol for? (I know you didn't start it -- I just never understood what it was for)
2. Why aren't these new properties part of it? :-)